go/ql/integration-tests/all-platforms/go/resolve-build-environment/newer-go-needed/test.py

```diff
@@ -1,3 +1,3 @@
-from go_integration_test import *
-
-go_integration_test(toolchain="go1.21.0")
+def test(codeql, go, check_build_environment):
+    check_build_environment.env = {"GOTOOLCHAIN": "go1.21.0"}
+    codeql.database.create(source_root="src")
```

Comment:
It's a bit weird that the `GOTOOLCHAIN` setting is for the build environment setting... I wonder if the `codeql.database.create` step brings anything to this test at all. @mbg maybe this test should be about the build environment only, dropping the database creation step altogether? I guess before we had the do-it-all `go_integration_test` that came as a package with both, but now we can select what to do

Reply:
> It's a bit weird that the `GOTOOLCHAIN` setting is for the build environment setting

It forces `go version` to return a particular version, which in turn makes it seem to `resolve build-environment` that this version is installed, forcing it to recommend a higher version based on the project's requirements. Concretely, here the project requires go1.22 and we make it look like only go1.21 is installed, so `resolve build-environment` should recommend to install go1.22. This is what the test is testing: the intended behaviour of `resolve build-environment`. 

> I wonder if the codeql.database.create step brings anything to this test at all.

Let's keep this please, as although this is not the primary goal of this particular test, having it also check that we can successfully extract the project does no harm.